src/modules/News/controller.ts

```diff
@@ -13,4 +13,41 @@ const getLatestNews = async (req: Request, res: Response) => {
   }
 };
 
-export default { getLatestNews };
+const getTopHeadlines = async (req: Request, res: Response) => {
+  const { country, category, limit } = req.query;
+  try {
+    const news = await fetchFromNewsAPI('/top-headlines', {
+      country,
+      category,
+      pageSize: limit,
+    });
+
+    const transformedArticles = news.articles.map(
+      (
+        article: {
+          title: unknown;
+          description: unknown;
+          url: unknown;
+          source: { name: unknown };
+          publishedAt: unknown;
+        },
+        index: number,
+      ) => ({
+        id: `news${index + 1}`,
```

Comment:
Suggestion: double check if NEWS API is returning an ID, if yes, return that.

Reply:
Checked wikipedia there is a standard for 2 character country codes